zerver/views/users.py

```diff
@@ -554,3 +554,28 @@ def get_subscription_backend(request: HttpRequest, user_profile: UserProfile,
     subscription_status = {'is_subscribed': subscribed_to_stream(target_user, stream_id)}
 
     return json_success(subscription_status)
+
+@has_request_variables
+def get_user_by_email(request: HttpRequest, user_profile: UserProfile, email: str,
+                      include_custom_profile_fields: bool=REQ(validator=check_bool,
+                                                              default=False),
+                      client_gravatar: bool=REQ(validator=check_bool, default=False),
+                      ) -> HttpResponse:
+    realm = user_profile.realm
+    if realm.email_address_visibility != Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+        # If email addresses are only available to administrators,
+        # clients cannot compute gravatars, so we force-set it to false.
+        client_gravatar = False
+    target_user = None
+    if email is not None:
+        target_user = get_user_by_delivery_email(email, realm)
```

Comment:
This should be by email, not `delivery_email`; the latter should only be used in contexts for sending emails, not API contexts exposed to other users (like this one).

Reply:
Ok ,I would make the change.